pkg/controllers/nodeclaim/termination/controller.go

```diff
@@ -84,15 +88,14 @@ func (c *Controller) Finalize(ctx context.Context, nodeClaim *v1beta1.NodeClaim)
 			}
 		}
 	}
-	// We wait until all the nodes associated with this nodeClaim have completed their deletion before triggering the finalization of the nodeClaim
-	if len(nodes) > 0 {
-		return reconcile.Result{}, nil
+	requeue, err := utils.EnsureTerminated(ctx, c.kubeClient, nodeClaim, c.cloudProvider)
+	if err != nil {
+		return reconcile.Result{}, err
 	}
-	if nodeClaim.Status.ProviderID != "" {
-		if err = c.cloudProvider.Delete(ctx, nodeClaim); cloudprovider.IgnoreNodeClaimNotFoundError(err) != nil {
-			return reconcile.Result{}, fmt.Errorf("terminating cloudprovider instance, %w", err)
-		}
+	if requeue {
+		return reconcile.Result{Requeue: requeue}, nil
```

Comment:
Same here. Should we be doing this with exponential back-off or be doing a static poll for rechecking the instance termination against the Cloudprovider